.gitlab-ci.yml

```diff
@@ -15,15 +15,15 @@ variables:
       aud: image-integrity
   script:
     - set -x
-    - builder=$(docker buildx create --use)
-    - trap "docker buildx rm $builder" EXIT
     # Construct valid --build-args arguments from the DOCKER_BUILD_ARGS variable
     - BUILD_ARGS=""; IFS=$'\n'; for arg in $DOCKER_BUILD_ARGS; do BUILD_ARGS+=" $(echo "--build-arg $arg")"; done; IFS=$' ';
     - IMAGE_TAG="$CI_COMMIT_TAG"
     - if [ "$TARGET" = "debug" ] ; then IMAGE_TAG="$IMAGE_TAG-debug" ; fi
     - IMAGE_REF="registry.ddbuild.io/$IMAGE_NAME:$IMAGE_TAG"
     - METADATA_FILE=$(mktemp)
-    - docker buildx build --platform linux/amd64,linux/arm64 --tag $IMAGE_REF --file $DOCKERFILE_PATH $BUILD_ARGS --label CILIUM_VERSION=$(cat VERSION) --label target=prod --target $TARGET --push --metadata-file $METADATA_FILE $DOCKER_CONTEXT
+    # Omit git information from the build
+    - sed -i '/^GO_BUILD_FLAGS =/a GO_BUILD_FLAGS += -buildvcs=false' Makefile.defs
```

Comment:
For the life of me I can't figure out why this is needed now and wasn't needed before.

Without `-buildvcs=false` all builds were failing with:
```
#36 [linux/amd64 builder 3/5] RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium     --mount=type=cache,target=/root/.cache     --mount=type=cache,target=/go/pkg     make GOARCH=amd64 RACE=${RACE} NOSTRIP=${NOSTRIP} NOOPT=${NOOPT} LOCKDEBUG=${LOCKDEBUG} cilium-operator     && mkdir -p /out/linux/amd64/usr/bin && mv cilium-operator /out/linux/amd64/usr/bin
#36 0.199 error: unable to normalize alternate object path: /opt/git-cache/cilium/objects
#36 0.199 fatal: bad object HEAD
#36 0.217 CGO_ENABLED=0 go build -mod=vendor -ldflags '-X "github.com/cilium/cilium/pkg/version.ciliumVersion=1.13.11 " -s -w -X "github.com/cilium/cilium/pkg/envoy.RequiredEnvoyVersionSHA=ad82c7c56e88989992fd25d8d67747de865c823b" -X "github.com/cilium/cilium/pkg/datapath/loader.DatapathSHA256=b28b3dc92a2fa9c128435db47a54ecc83b1b872da962c8300c5a87db32e17197" ' -tags=osusergo,ipam_provider_aws,ipam_provider_azure,ipam_provider_operator,ipam_provider_alibabacloud  -o cilium-operator
#36 2.166 error obtaining VCS status: exit status 128
#36 2.166 	Use -buildvcs=false to disable VCS stamping.
```
I spent a bit of time on this problem but still unsure how to fix it for real, it is coming from somewhere deep in Go.

You can still see
```
#35 0.171 error: object directory /opt/git-cache/cilium/objects does not exist; check .git/objects/info/alternates
#35 0.171 fatal: bad object HEAD
```
In the build logs now but it's no longer failing so the issue must be somewhere behind that "exit status 128" 🤷 
Luckily, we can disable VCS stamping with Go 1.18+ (https://tip.golang.org/doc/go1.18#go-command) and I don't think we really need it anyway.

Also this `sed` command is the easiest way I found to override this option for all builds, it's not pretty but it works.